gcloud-java-dns/src/main/java/com/google/gcloud/dns/testing/OptionParsers.java

```diff
@@ -0,0 +1,256 @@
+          break;
+        default:
+          break;
+      }
+    }
+    return managedZone;
+  }
+
+  static Change extractFields(Change fullChange, String... fields) {
+    if (fields == null || fields.length == 0) {
+      return fullChange;
+    }
+    Change change = new Change();
+    for (String field : fields) {
+      switch (field) {
+        case "additions":
+          change.setAdditions(fullChange.getAdditions());
+          break;
+        case "deletions":
+          change.setDeletions(fullChange.getDeletions());
+          break;
+        case "id":
+          change.setId(fullChange.getId());
+          break;
+        case "startTime":
+          change.setStartTime(fullChange.getStartTime());
+          break;
+        case "status":
+          change.setStatus(fullChange.getStatus());
+          break;
+        default:
+          break;
+      }
+    }
+    return change;
+  }
+
+  static Project extractFields(Project fullProject, String... fields) {
+    if (fields == null || fields.length == 0) {
+      return fullProject;
+    }
+    Project project = new Project();
+    for (String field : fields) {
+      switch (field) {
+        case "id":
+          project.setId(fullProject.getId());
+          break;
+        case "number":
+          project.setNumber(fullProject.getNumber());
+          break;
+        case "quota":
+          project.setQuota(fullProject.getQuota());
+          break;
+        default:
+          break;
+      }
+    }
+    return project;
+  }
+
+  static ResourceRecordSet extractFields(ResourceRecordSet fullRecord, String... fields) {
+    if (fields == null) {
+      return fullRecord;
+    }
+    ResourceRecordSet record = new ResourceRecordSet();
+    for (String field : fields) {
+      switch (field) {
+        case "name":
+          record.setName(fullRecord.getName());
+          break;
+        case "rrdatas":
+          record.setRrdatas(fullRecord.getRrdatas());
+          break;
+        case "type":
+          record.setType(fullRecord.getType());
+          break;
+        case "ttl":
+          record.setTtl(fullRecord.getTtl());
+          break;
+        default:
+          break;
+      }
+    }
+    return record;
+  }
+
+  static Map<String, Object> parseListChangesOptions(String query) {
+    Map<String, Object> options = new HashMap<>();
+    if (query != null) {
+      String[] args = query.split("&");
+      for (String arg : args) {
+        String[] argEntry = arg.split("=");
+        switch (argEntry[0]) {
+          case "fields":
+            // todo we do not support fragmentation in deletions and additions in the library
+            String replaced = argEntry[1].replace("changes(", ",").replace(")", ",");
+            options.put("fields", replaced.split(",")); // empty strings will be ignored
+            break;
+          case "pageToken":
+            options.put("pageToken", argEntry[1]);
+            break;
+          case "sortBy":
```

Comment:
If you provide an invalid value and call the service, it throws an error. It is here only to make validation possible to simulate this behaviour.